c_src/exml.cpp

```diff
@@ -67,6 +67,34 @@ ERL_NIF_TERM atom_pretty;
 ERL_NIF_TERM atom_true;
 constexpr const unsigned char EMPTY[1] = {0};
 
+class Version {
+public:
+  Version(const std::string str) {
+    for (std::size_t i = 0; i < str.size();) {
```

Comment:
I feel `old` reading this modern C++ code and having difficulties to understand it easily.

Reply:
![i_hug_that_feel](https://user-images.githubusercontent.com/548816/40348753-d5d0983e-5da4-11e8-9559-d366211faae6.png)

Reply:
@michalwski The strings are saved in RapidXML document's memory pool. We set the memory pool up to initial size of 10MB, and clear the pool before use in each NIF call. To have any impact on memory at all, the strings in a single `#xmlel{}` would have to take more than those 10MB.